examples/example6.go

```diff
@@ -0,0 +1,112 @@
+// For example, your request has a "/order/{random}" path
+// and we want to rewrite it to /order/1234 so we can match it later.
+// We change the response status code.
+// We add headers based on request method.
+func Example6() {
+	cfg := govcr.VCRConfig{
+		Logging: true,
+	}
+
+	// The filter will neutralize a value in the URL.
+	// In this case we rewrite /order/{random} to /order/1234
+	replacePath := govcr.RequestFilter(func(req govcr.Request) govcr.Request {
+		// Replace path with a predictable one.
+		req.URL.Path = "/order/1234"
+		return req
+	})
+	// Only execute when we match path.
+	replacePath = replacePath.OnPath(`example\.com\/order\/`)
+
+	// Add to request filters.
+	cfg.RequestFilters.Add(replacePath)
+	cfg.RequestFilters.Add(govcr.RequestDeleteHeaderKeys("X-Transaction-Id"))
+
+	// Add filters
+	cfg.ResponseFilters.Add(
+		// Always transfer 'X-Transaction-Id' as in example 5.
+		govcr.ResponseTransferHeaderKeys("X-Transaction-Id"),
+
+		// Change status 404 to 202.
+		func(resp govcr.Response) govcr.Response {
+			if resp.StatusCode == http.StatusNotFound {
+				resp.StatusCode = http.StatusAccepted
+			}
+			return resp
+		},
+
+		// Add header if method was "GET"
+		govcr.ResponseFilter(func(resp govcr.Response) govcr.Response {
+			resp.Header.Add("method-was-get", "true")
+			return resp
+		}).OnMethod(http.MethodGet),
+
+		// Add header if method was "POST"
+		govcr.ResponseFilter(func(resp govcr.Response) govcr.Response {
+			resp.Header.Add("method-was-post", "true")
+			return resp
+		}).OnMethod(http.MethodPost),
+
+		// Add actual request URL to header.
+		govcr.ResponseFilter(func(resp govcr.Response) govcr.Response {
+			url := resp.Request().URL
+			resp.Header.Add("get-url", url.String())
+			return resp
+		}).OnMethod(http.MethodGet),
+	)
+
+	orderID := fmt.Sprint(rand.Int63())
+	vcr := govcr.NewVCR(example6CassetteName, &cfg)
+
+	// create a request with our custom header and a random url part.
+	req, err := http.NewRequest("POST", "http://www.example.com/order/"+orderID, nil)
+	if err != nil {
+		fmt.Println(err)
+	}
+	runRequest(req, vcr)
+
+	// create a request with our custom header and a random url part.
+	req, err = http.NewRequest("GET", "http://www.example.com/order/"+orderID, nil)
+	if err != nil {
+		fmt.Println(err)
+	}
+	runRequest(req, vcr)
+
+}
+
+func runRequest(req *http.Request, vcr *govcr.VCRControlPanel) {
+	req.Header.Add("X-Transaction-Id", time.Now().String())
+	// run the request
+	resp, err := vcr.Client.Do(req)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	// verify outcome
+	if req.Header.Get("X-Transaction-Id") != resp.Header.Get("X-Transaction-Id") {
+		fmt.Println("Header transaction Id verification failed - this would be the live request!")
+	} else {
+		fmt.Println("Header transaction Id verification passed - this would be the replayed track!")
+	}
+
+	// print outcome.
+	fmt.Println("Status code:", resp.StatusCode, " (should be 404 on real and 202 on replay)")
```

Comment:
This should form the basis of a formal go example test (as in the root folder `xxx_examplen_test.go` and some unit tests too.